docs/DeveloperGuide.md

```diff
@@ -277,6 +277,51 @@ Step 3. The GUI displays the list of companies filtered by the given search term
 
 ---
 
+### Favourite/unfavourite feature
+
+#### Implementation
+
+The `fav`/`unfav` command allows users to mark specified a `Company` as favourites or not. Companies that are favourited
+will appear at the top of the company list at all times along with other favourited companies (provided that they are not hidden by other commands like `find`).
+
+The `fav` command is facilitated by `FavCommand` and `FavCommandParser`. Similarly, `unfav` command is facilitated by
+`UnfavCommand` and `UnfavCommandParser`. For brevity, we refer to both using `XXFavCommand` and `XXFavCommanParser`
+Since they function similarly.
+
+The following methods and operations are involved:
+* `XXFavCommand#execute(Model model)` &mdash; (Un)Favourites the specified `Company`.
+* `Model#setCompany(Company targetCompany, Company updatedCompany` &mdash; Replaces the existing company in the model with
+the same `Company` but with its `IsFavourite` set to `true` or `false` depending on `XXFavCommand`
+* `AddressBook#setCompany(Company target, Company editedCompany` &mdash; Commits the changes to the address book.
+* `UniqueCompanyList#setCompany(Company target, Company editedCompany)` &mdash; Updates internal list, sorting by favourited companies.
+
+#### Example usage scenario:
+
+Step 1. The user selects a company and executes the `XXfav` command with the corresponding company list index. 
+
+Step 2. The `XXFavCommand` fetches the selected company and creates an identical `Company` but with the updated `IsFavourite` field and
+calls `Model#SetCompany()` to replace the old company in the model with the updated one.
+
+Step 3. The changes are committed to the address book by calling `AddressBook#SetCompany()`; the address book replaces
+the old company in its `UniqueCompanyList` sorting by `IsFavourite` in the process by calling `UniqueCompanyList#SetCompany`.
```

Comment:
Similar to previous concerns for AddressBook#setCompany and UniqueCompanyList#setCompany